### PR TITLE
link: fix memory leaks

### PR DESCRIPTION
**Pull request by jacobly0**
* Fix linker memory leaks found while running `zig build test-cases`.
 * Add missing target to test manifest.

### CODE REVIEW

File: src/link/Coff.zig [hunk 1]
```diff
@@ -1489,9 +1492,8 @@ pub fn getGlobalSymbol(self: *Coff, name: []const u8) !u32 {
     gop.value_ptr.* = sym_loc;
 
     const gpa = self.base.allocator;
-    const sym_name = try gpa.dupe(u8, name);
     const sym = self.getSymbolPtr(sym_loc);
-    try self.setSymbolName(sym, sym_name);
+    try self.setSymbolName(sym, name);
```
Comment (kubkon) [inline src/link/Coff.zig:1496]: Nice catch, thanks!

File: src/link/MachO.zig [hunk 2]
```diff
@@ -2576,10 +2576,14 @@ pub fn deleteExport(self: *MachO, exp: Export) void {
 }
 
 fn freeRelocationsForAtom(self: *MachO, atom: *Atom) void {
-    _ = self.relocs.remove(atom);
-    _ = self.rebases.remove(atom);
-    _ = self.bindings.remove(atom);
-    _ = self.lazy_bindings.remove(atom);
+    var removed_relocs = self.relocs.fetchRemove(atom);
```
Comment (kubkon) [inline src/link/MachO.zig:2579]: Great catch, thanks!

**Review by kubkon — APPROVED** [review]
Looks awesome, thanks @jacobly0! What I am surprised about, we used to report these when running `test-cases` but this no longer happens it seems. Has something changed recently? Have we ditched GPA in favour of C alloc?

**Comment by kubkon** [comment]
Hmm, seems like you managed to get memory leak with `test-cases` while I don't see anything when running locally on an M1 (except CBE failures todo with child process exec but that's unrelated I guess).

**Comment by jacobly0** [comment]
> Hmm, seems like you managed to get memory leaks report with `test-cases` while I don't see anything when running locally on an M1 (except CBE failures todo with child process exec but that's unrelated I guess).

I'm not entirely sure how I'm triggering it, but maybe try `zig build test-cases -Dforce-gpa=true`?

**Comment by Luukdegram** [comment]
The compiler uses the `c_allocator` whenever libc is linked. Meaning that whenever the LLVM backend is enabled, or when you're on a platform where we always link with libc, the compiler does not use the GPA, unless you force it with the flag Jacobly mentioned. I believe we previously had that flag set for one of the CI's to ensure we didn't leak anything, but that was probably lost.

**Comment by kubkon** [comment]
Ohhh, right, that's it, thanks! In that case, I think it might be worth re-adding that flag to one of the CI's to avoid this type of regressions in the future (what do you think @andrewrk?), and I will remember to run the tests locally with `-Dforce-gpa=true` from now on as most of the leaks are my fault 😅

**Comment by andrewrk** [comment]
Sure sounds like a good idea, you know what they say about proper behavior - [if you like it then you shoulda put a test on it](https://www.youtube.com/watch?v=4m1EFMoRFvY)!

**Pull request merged**